x-pack/filebeat/tests/integration/managerV2_test.go

```diff
@@ -500,6 +528,157 @@ func TestRecoverFromInvalidOutputConfiguration(t *testing.T) {
 	}
 }
 
+func TestAgentPackageVersionOnStartUpInfo(t *testing.T) {
+	wantVersion := "8.13.0+build20131123"
+
+	filebeat := integration.NewBeat(
+		t,
+		"filebeat",
+		"../../filebeat.test",
+	)
+
+	logFilePath := filepath.Join(filebeat.TempDir(), "logs-to-ingest.log")
+	generateLogFile(t, logFilePath)
+
+	eventsDir := filepath.Join(filebeat.TempDir(), "ingested-events")
+	logLevel := proto.UnitLogLevel_INFO
+	units := []*proto.UnitExpected{
+		{
+			Id:             "output-file-unit",
+			Type:           proto.UnitType_OUTPUT,
+			ConfigStateIdx: 1,
+			State:          proto.State_HEALTHY,
+			LogLevel:       logLevel,
+			Config: &proto.UnitExpectedConfig{
+				Id:   "default",
+				Type: "file",
+				Name: "events-to-file",
+				Source: integration.RequireNewStruct(t,
+					map[string]interface{}{
+						"name": "events-to-file",
+						"type": "file",
+						"path": eventsDir,
+					}),
+			},
+		},
+		{
+			Id:             "input-unit-1",
+			Type:           proto.UnitType_INPUT,
+			ConfigStateIdx: 1,
+			State:          proto.State_HEALTHY,
+			LogLevel:       logLevel,
+			Config: &proto.UnitExpectedConfig{
+				Id:   "filestream-monitoring-agent",
+				Type: "filestream",
+				Name: "filestream-monitoring-agent",
+				Streams: []*proto.Stream{
+					{
+						Id: "log-input-1",
+						Source: integration.RequireNewStruct(t, map[string]interface{}{
+							"enabled": true,
+							"type":    "log",
+							"paths":   []interface{}{logFilePath},
+						}),
+					},
+				},
+			},
+		},
+	}
+	agentInfo := proto.AgentInfo{
+		Id:       "agent-id",
+		Version:  wantVersion,
+		Snapshot: true,
+	}
+
+	server := &mock.StubServerV2{
+		CheckinV2Impl: func(observed *proto.CheckinObserved) *proto.CheckinExpected {
+			return &proto.CheckinExpected{
+				AgentInfo: &agentInfo,
```

Comment:
I wanted to see if the test depended on receiving agent info from the checkin expected message instead of the start up info on stdin. 

Making this change the test fails:

```diff
diff --git a/x-pack/filebeat/tests/integration/managerV2_test.go b/x-pack/filebeat/tests/integration/managerV2_test.go
index 1b203d2ee5..5144b3cd43 100644
--- a/x-pack/filebeat/tests/integration/managerV2_test.go
+++ b/x-pack/filebeat/tests/integration/managerV2_test.go
@@ -593,8 +593,12 @@ func TestAgentPackageVersionOnStartUpInfo(t *testing.T) {
        server := &mock.StubServerV2{
                CheckinV2Impl: func(observed *proto.CheckinObserved) *proto.CheckinExpected {
                        return &proto.CheckinExpected{
-                               AgentInfo: &agentInfo,
-                               Units:     units,
+                               AgentInfo: &proto.AgentInfo{
+                                       Id: "agent-id",
+                                       // Version:  wantVersion,
+                                       Snapshot: true,
+                               },
+                               Units: units,
                        }
                },
                ActionImpl: func(response *proto.ActionResponse) error { return nil },
```

That is because of https://github.com/elastic/elastic-agent-client/blob/17d2b257ff148cd86de88252359bef0f28df9ab8/pkg/client/client_v2.go#L571-L581

```go
// applyExpected syncs the expected state with the current state.
func (c *clientV2) applyExpected(expected *proto.CheckinExpected) {
	if expected.AgentInfo != nil {
		c.agentInfoMu.Lock()
		c.agentInfo = &AgentInfo{
			ID:       expected.AgentInfo.Id,
			Version:  expected.AgentInfo.Version,
			Snapshot: expected.AgentInfo.Snapshot,
		}
		c.agentInfoMu.Unlock()
	}
```

If we never expect the agent version to change, do we need to update it on each expected message? Does that let us simplify the client by removing the agentInfoMu lock?

This is not strictly related to this PR, but the series of changes involved.

Reply:
you're right. Now we're sending the whole agent info at start up it's pointless to send it during check-in. Besides the beats ignore the new version, even if the agent sends another, the beats will not apply it and keep reporting the one received at start up